README.md

```diff
@@ -4,35 +4,47 @@
 
 ## Building the CLI
 
-If you simply want to build a binary locally, run
-```sh
-$ make build/turnkey
-```
-
-We use [GoReleaser](https://goreleaser.com/) to build and release our binaries.
-
-To build a release locally:
+If you simply want to build binaries locally, run
 ```
-$ goreleaser release --snapshot --rm-dist
+$ make
 ```
 
 To release:
 ```
-# Generate a Github token with "write:packages"
-# ==> https://github.com/settings/tokens
-$ export GITHUB_TOKEN=<your token>
-$ git tag -s -a vx.y.z -m "New release: x.y.z" # create a signed tag
-$ git tag -v vx.y.z # verify the tag
-$ git push origin vx.y.z
-$ goreleaser release --rm-dist
+make dist
 ```
 
 ## Installing the CLI
 
-```
-brew tap tkhq/tap
-brew install turnkey
-```
```

Comment:
Is there a way to keep this brew path for install and verify the installed binaries as a post-install step? A lot of people won't have the patience to clone and verify, and I'm guessing a lot of people won't have gpg setup correctly or won't trust the signatures by default. I'm also not sure this flow is realistic for automated build pipelines (what if I simply want `turnkey` availble in my integration tests?)

It'd be nice to release the binaries and integrate with a package manager. I know you hate brew with a passion but that's what most of the devs on MacOSX use. Maybe we force another, more secure package manager on them? (MacPorts?)

Reply:
> It'd be nice to release the binaries and integrate with a package manager. 

I fully agree, and hope we can see the turnkey binaries released in package managers that support reproducible builds and signing, such as Arch or Debian, but with good reason it takes time to earn a reputation and the trust of external maintainers to include it. We have to provide a best effort path for people with high security needs to form trust in binaries directly from us in advance of that.

> I know you hate brew with a passion but that's what most of the devs on MacOSX use. 

A lot of the security practices we provide like enclave based signing and webauthn request signing are not something most devs would know to ask for, as most devs lack a background in security. I have hope that if we are the ones that help people understand their threat model many will chose us over competing solutions that don't offer any options for high risk applications.

> Maybe we force another, more secure package manager on them? (MacPorts?)

Macports is slightly better, in that they bother to sign packages at all, though it is bulk automated signing so it is of limited value. The more important thing is they don't just let any randos push what they want like brew or npm, but the other edge of that sword is like linux distros you would have to go through the process to become a volunteer maintainer first and get the package approved which could prove challenging as it does not have a reputation others would recognize yet.

> I'm guessing a lot of people won't have gpg setup correctly or won't trust the signatures by default.

They should not trust the signatures by default, and that is a good thing. Encouraging people to stop and think about what they are doing is more important for their security than any engineering we can do.

gpg does ship with macos and virtually all linux distros. It also ships able to verify signatures with no configuration required.

> Is there a way to keep this brew path for install and verify the installed binaries as a post-install step?

I do not feel we should encourage use of unsigned package managers like brew on systems intended to participate in high accountability cryptography operations, which is all Turnkey use cases. That said if you feel the only way some people will kick the tires on Turnkey is with familiar low security paths, then a compromise would be providing such options with clear disclaimers.

We could then break up the installation paths broken out from easy but low security to difficult but high security.

Something like:

- low security
  - These options are provided only for convenience and familiarity
  - These options have no verification, so you could be downloading malware
  - We strongly encourage you to not use these methods on systems you need to be able to trust
  - brew ...
  - curl | bash ...
- medium security
  - Verify signatures on your binaries to ensure multiple people signed them
  - Verify the identities of some or all of the signers and that they are using well known keys
  - ...
- High security
  - Compile the binaries yourself and ensure they match the ones we publish
  - ...
  
If you agree with this path I can adjust the Readme along these lines, re-adding the brew install path as well as a bash one liner.

> I'm also not sure this flow is realistic for automated build pipelines (what if I simply want turnkey available in my integration tests?)

Automated build pipelines can generally git-clone anything to get easy access to the binaries in the dist directory, but the bash one liner could also exist for low security use cases like git hook integration tests.

Reply:
> We could then break up the installation paths further from easy but low security to difficult but high security.

I like this 👍 

> Automated build pipelines can generally git-clone anything to get easy access to the binaries in the dist directory, but the bash one liner could also exist for low security use cases like git hook integration tests.

Makes sense. I also think we should provide a simple option to download and shasum check against known values we publish on our turnkey domain? We do this for e.g. awscli. Not the same as reproducible build, but it's a good option to ensure release binary integrity.

Reply:
> They should not trust the signatures by default, and that is a good thing

Might be worth including steps to import keys in gpg such that make verify succeeds? I'm concerned about "expecting" `make verify` to fail. Users might think something's off with our releases instead of looking at the state of their keystore (see e.g. https://github.com/bitcoin-core/bitcoincore.org/issues/793)

Reply:
Make verify should succeed. I mean to say that it just will succeed but with keys with no trust history. We could output where people could go to verify the key fingerprints though.

> Makes sense. I also think we should provide a simple option to download and shasum check against known values we publish on our turnkey domain? We do this for e.g. awscli. Not the same as reproducible build, but it's a good option to ensure release binary integrity.

This means updating the website every time we make a new release. I would rather teach people to use and recognize that there should be multiple recognized signatures in the "make verify" output.

If we have a "team" page on the website, linking to the key fingerprints and keyoxide profiles of each engineer would allow people to form initial trust that given commit/release signature belong to members of our team